dist/threads/threads.xs

```diff
@@ -667,8 +668,6 @@ S_ithread_run(void * arg)
     MUTEX_UNLOCK(&thread->mutex);
     MUTEX_UNLOCK(&MY_POOL.create_destruct_mutex);
 
-    thread_locale_term();
```

Comment:
I think this is wrong, or if it's right it needs more explanation of why it's right.

`S_ithread_clear` will be called from other (OS) threads, whereas S_ithread_run will be run from the interpreter's own thread. This is an essential distinction.